src/main/java/frc/robot/subsystems/Vision.java

```diff
@@ -68,8 +70,20 @@
 
   private boolean aimingLight;
   private boolean shootingLight;
+
+  private PIDController pid;
+  private double p = 0.02;
+  private double i = 0.02;
+  private double d = 0.002;
```

Comment:
I learned about PID in school but I'll be honest I don't really remember the details.  Just curious, how did you pick these values?

Reply:
I tested the auto align during lunch today and these values gave the best results.

I started by setting the I and D values to 0 and changing P until the robot oscillated around the target. Then I set P to be half that value. Then I set I so that the robot still aligned itself with the target relatively quickly, and then I increased D until the motion looked smooth enough. I pretty much just followed [this guide](https://pidexplained.com/how-to-tune-a-pid-controller/).